azurerm/resource_arm_netapp_account.go

```diff
@@ -0,0 +1,288 @@
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for present of existing NetApp Account %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_netapp_account", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	activeDirectories := d.Get("active_directory").([]interface{})
+	t := d.Get("tags").(map[string]interface{})
+
+	accountParameters := netapp.Account{
+		Location: utils.String(location),
+		AccountProperties: &netapp.AccountProperties{
+			ActiveDirectories: expandArmNetAppActiveDirectories(activeDirectories),
+		},
+		Tags: tags.Expand(t),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, accountParameters, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error creating NetApp Account %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of NetApp Account %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving NetApp Account %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read NetApp Account %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmNetAppAccountRead(d, meta)
+}
+
+func resourceArmNetAppAccountRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.AccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["netAppAccounts"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] NetApp Accounts %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading NetApp Accounts %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if props := resp.AccountProperties; props != nil {
+		if err := d.Set("active_directory", flattenArmNetAppActiveDirectories(props.ActiveDirectories)); err != nil {
+			return fmt.Errorf("Error setting `active_directory`: %+v", err)
+		}
+	}
+
+	currentTags := make(map[string]*string)
+	if v := resp.Tags; v != nil {
+		tagMap := v.(map[string]interface{})
+
+		for k, v := range tagMap {
+			currentTags[k] = utils.String(v.(string))
+		}
+	}
```

Comment:
It seems that we cannot do this since the type of resp.Tags is "Tags interface{} `json:"tags,omitempty"`" and function tags.FlattenAndSet only can accept type "tagMap map[string]*string". If I write code as "return tags.FlattenAndSet(d, resp.Tags)" and it would throw below error message.
![image](https://user-images.githubusercontent.com/19754191/66373687-cb558580-e9db-11e9-8565-0b08cbe58c5a.png)

Reply:
tbf, this is probably an error/issue with the swagger and API and should be tracked down. The swagger is probably missing the type sois defaulting to an interface rather then the correct types